### PR TITLE
Add tests for Falcon 3.0.0

### PR DESCRIPTION
**Pull request by brettlangdon**
## Description
Falcon 3.0.0 was released in April. This PR moves Falcon from tox to riot and updates tested versions to include 3.x and "latest".

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Comment by brettlangdon** [comment]
TODO: Falcon 3.x moved from `app = falcon.API()` to `app = falcon.App()` we need to update our instrumentation and tests.

While this is working fine for those who are still using `falcon.API()` it does not work for those using the new `falcon.App()`

File: riotfile.py [hunk 1]
```diff
@@ -180,6 +180,80 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
...
+                # Python 3.5+
+                Venv(
+                    pys=select_pys(min_version="3.5"),
+                    pkgs={
+                        "falcon": [
+                            "~=2.0.0",
+                            "~=2.0",  # latest 2.x
+                        ]
+                    },
+                ),
```
Comment (Yun-Kim) [inline riotfile.py:219]: What's the purpose of having this extra venv?

File: riotfile.py [hunk 2]
```diff
@@ -180,6 +180,80 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
...
+                Venv(
+                    pys=select_pys(min_version="3.5"),
+                    pkgs={
+                        "falcon": [
+                            "~=3.0.0",
+                            "~=3.0",  # latest 3.x
+                            latest,
+                        ]
+                    },
+                ),
```
Comment (Yun-Kim) [inline riotfile.py:242]: Same here, not sure how exactly riot/tox works so just curious if it's necessary since this doesn't actually run any commands

**Review by Yun-Kim — APPROVED** [review]
A couple questions but otherwise LGTM

File: riotfile.py (hunk 2 above)
Comment (brettlangdon) [reply riotfile.py:242]: it'll inherit the base command from the parent venv. I can move the command into these to make it more clear if that'll help?

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**